show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/screens/CoordinatesLayout.kt

```diff
@@ -0,0 +1,137 @@
+package com.esri.arcgismaps.sample.showcoordinatesinmultipleformats.screens
+
+import android.content.res.Configuration
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.WindowInsets
+import androidx.compose.foundation.layout.ime
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberUpdatedState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.focus.focusRequester
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.core.view.WindowCompat
+import com.esri.arcgismaps.sample.showcoordinatesinmultipleformats.components.MapViewModel
+
+@Composable
+fun CoordinatesLayout(
+    modifier: Modifier = Modifier,
+    onQuerySubmit: (MapViewModel.NotationType, String) -> Unit,
+    decimalDegrees: MutableState<String>,
+    degreesMinutesSeconds: MutableState<String>,
+    utm: MutableState<String>,
+    usng: MutableState<String>
```

Comment:
I understand now, you made these public `MutableState` in the VM so they could be modified here, but I still think it is a bit of a code smell. I would normally expect that composables [cannot modify any state coming in, but would instead be passed a function](https://developer.android.com/jetpack/compose/architecture#udf) which calls down to the VM which then changes the state from inside the VM. This is what you normally see in the docs for [eg. `TextField`](https://developer.android.com/jetpack/compose/text/user-input#choose-textfield).

I'm not sure what the best way to combine simplicity and standard patterns is here. Maybe a text field composable which takes in just one string and lambda to keep things clean, something like:

```kotlin
@Composable
fun CoordinatesLayout(
    modifier: Modifier = Modifier,
    mapViewModel: MapViewModel
) {
    CoordinateField(
        notationType = MapViewModel.NotationType.DD,
        text = mapViewModel.decimalDegrees,
        onTextChanged = mapViewModel::setDecimalDegrees,
        onQuerySubmit = mapViewModel::fromCoordinateNotationToPoint
    )
...
}

@Composable
fun CoordinateField(
    modifier: Modifier = Modifier,
    notationType: MapViewModel.NotationType
    text: String = "",
    onTextChanged: (String) -> Unit = {},
    onQuerySubmit: (MapViewModel.NotationType, String) -> Unit = {}
) {
      OutlinedTextField(
          modifier = modifier,
          value = text,
          maxLines = 1,
          onValueChange = onTextChanged,
          label = { Text("Decimal Degrees") },
          keyboardOptions = keyboardOptions,
          keyboardActions = KeyboardActions(
              onSearch = {
                  onQuerySubmit(notationType, text)
                  focusManager.clearFocus()
              },
          ),
      )
}
```

Reply:
Really nice breaking down the `CoordinatesLayout` using `CoordinateField` into it's own Compose functions. I'm beginning to understand the concept of not modifying any state coming in from the VM, but instead passing the new state as a function. This would require us to pass the instance of the `mapViewModel` to every new compose function, which should be fine. 

One point to note, I couldn't name the setter function as `setDecimalDegrees` since it would conflict with the name used by the JVM as it would throw an error:
```console
Platform declaration clash: The following declarations have the same JVM signature (setDecimalDegrees(Ljava/lang/String;)V):
    fun `<set-decimalDegrees>`(`<set-?>`: String): Unit defined in com.esri.arcgismaps.sample.showcoordinatesinmultipleformats.components.MapViewModel
    fun setDecimalDegrees(inputString: String): Unit defined in com.esri.arcgismaps.sample.showcoordinatesinmultipleformats.components.MapViewModel
```
Instead I chose to name it `setDecimalDegreesCoordinate` and similarly for the other functions.